packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -66,83 +73,70 @@ export interface TabPanelProps extends Omit<AriaTabPanelProps, 'children' | 'sty
-  flexBasis: '[0%]',
-  minHeight: 0,
-  minWidth: 0
-}, getAllowedOverrides({height: true}));
-
-export function TabPanel(props: TabPanelProps) {
-  return (
-    <AriaTabPanel
-      {...props}
-      style={props.UNSAFE_style}
-      className={(props.UNSAFE_className || '') + tabPanel(null, props.styles)} />
-  );
-}
-
-const tab = style({
-  ...focusRing(),
+const tabs = style({
   display: 'flex',
-  color: {
-    default: 'neutral-subdued',
-    isSelected: 'neutral',
-    isHovered: 'neutral-subdued',
-    isDisabled: 'disabled',
-    forcedColors: {
-      isSelected: 'Highlight',
-      isDisabled: 'GrayText'
-    }
-  },
-  borderRadius: 'sm',
-  gap: 'text-to-visual',
-  height: {
-    density: {
-      compact: 32,
-      regular: 48
-    }
-  },
-  alignItems: 'center',
-  position: 'relative',
-  cursor: 'default',
-  flexShrink: 0,
-  transition: 'default'
-}, getAllowedOverrides());
-
-const icon = style({
   flexShrink: 0,
-  '--iconPrimary': {
-    type: 'fill',
-    value: 'currentColor'
+  font: 'ui',
+  flexDirection: {
+    orientation: {
+      horizontal: 'column'
+    }
   }
-});
+}, getAllowedOverrides({height: true}));
 
-export function Tab(props: TabProps) {
-  let {density} = useSlottedContext(TabsContext) ?? {};
+function Tabs(props: TabsProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, TabsContext);
+  let {
+    density = 'regular',
+    isDisabled,
+    disabledKeys,
+    orientation = 'horizontal',
+    iconOnly = false
+  } = props;
+  let domRef = useDOMRef(ref);
+  let [value, setValue] = useControlledState(props.selectedKey, props.defaultSelectedKey ?? null!, props.onSelectionChange);
 
   return (
-    <RACTab
-      {...props}
-      style={props.UNSAFE_style}
-      className={renderProps => (props.UNSAFE_className || '') + tab({...renderProps, density}, props.styles)}>
-      <Provider
-        values={[
-          [TextContext, {styles: style({order: 1})}],
-          [IconContext, {
-            render: centerBaseline({slot: 'icon', styles: style({order: 0})}),
-            styles: icon
-          }]
-        ]}>
-        {typeof props.children === 'string' ? <Text>{props.children}</Text> : props.children}
-      </Provider>
-    </RACTab>
+    <Provider
+      values={[
+        [InternalTabsContext, {
+          density,
+          isDisabled,
+          orientation,
+          disabledKeys,
+          selectedKey: value,
+          onSelectionChange: setValue,
+          iconOnly
+        }]
+      ]}>
+      <CollapsingCollection containerRef={domRef}>
+        <RACTabs
+          {...props}
+          ref={domRef}
+          selectedKey={value}
+          onSelectionChange={setValue}
+          style={props.UNSAFE_style}
+          className={renderProps => (props.UNSAFE_className || '') + tabs({...renderProps}, props.styles)}>
+          <Provider
+            values={[
+              [TabsContext, {density, isDisabled, disabledKeys, orientation}]
```

Comment:
Do we need this context provider anymore? Looks like we use InternalTabsContext only and the props we may get from an external TabsContext would get passed down to RACTabs directly